cdc: add some explanations about lossy DDL (#13552)
pingcap/docs: ticdc/ticdc-faq.md

4. Create a new changefeed and start the replication task from `BackupTS`.
5. Delete the old changefeed.

> ## Does TiCDC synchronize data changes caused by lossy DDL operations to the downstream?

Reviewer: ```suggestion
## Does TiCDC replicate data changes caused by lossy DDL operations to the downstream?
```

Lossy DDL refers to DDL that might cause data changes when executed in TiDB. Some common lossy DDL operations include:

> - Modify the type of a column, for example: INT -> VARCHAR
> - Modify the length of a column, for example: VARCHAR(20) -> VARCHAR(10)
> - Modify the precision of a column, for example: DECIMAL(10, 3) -> DECIMAL(10, 2)
> - Modify the UNSIGNED or SIGNED attribute of a column, for example: INT UNSIGNED -> INT SIGNED

Reviewer: ```suggestion
- Modifying the type of a column, for example: INT -> VARCHAR
- Modifying the length of a column, for example: VARCHAR(20) -> VARCHAR(10)
- Modifying the precision of a column, for example: DECIMAL(10, 3) -> DECIMAL(10, 2)
- Modifying the UNSIGNED or SIGNED attribute of a column, for example: INT UNSIGNED -> INT SIGNED
```

> Before TiDB v7.1.0, TiCDC synchronizes DML events with identical old and new data to the downstream. When the downstream is MySQL, these DML events do not cause any data changes until the downstream receives and executes the DDL statement. However, when the downstream is Kafka or cloud storage, TiCDC writes a row of redundant data to the downstream.

Reviewer: ```suggestion
Before TiDB v7.1.0, TiCDC replicates DML events with identical old and new data to the downstream. When the downstream is MySQL, these DML events do not cause any data changes until the downstream receives and executes the DDL statement. However, when the downstream is Kafka or a cloud storage service, TiCDC writes a row of redundant data to the downstream.
```

> ## Will TiCDC synchronize data changes caused by lossy DDL to downstream?
> 
> Lossy DDL refers to DDL that may cause data changes when executed in TiDB. Some common lossy DDL operations include:
> 
> - Modify the type of the column, for example: INT -> VARCHAR
> - Modify the length of the column, for example: VARCHAR(20) -> VARCHAR(10)
> - Modify the precision of the column, for example: DECIMAL(10, 3) -> DECIMAL(10, 2)
> - Modify the sign of the column (signed/unsigned), for example: INT UNSIGNED -> INT SIGNED
> 
> Before TiDB v7.1.0, TiCDC would synchronize DML events with new and old data identical to downstream. When downstream is MySQL, these DML events will not cause any data changes. Only when downstream receives and executes the DDL statement will data change. However, when downstream is Kafka or cloud storage, TiCDC will write a useless piece of data to downstream.
> 
> Starting from TiDB v7.1.0, TiCDC will filter out these useless DML events and no longer synchronize them to downstream.

Reviewer 2: ```suggestion
## Does TiCDC synchronize data changes caused by lossy DDL operations to the downstream?

Lossy DDL refers to DDL that might cause data changes when executed in TiDB. Some common lossy DDL operations include:

- Modify the type of a column, for example: INT -> VARCHAR
- Modify the length of a column, for example: VARCHAR(20) -> VARCHAR(10)
- Modify the precision of a column, for example: DECIMAL(10, 3) -> DECIMAL(10, 2)
- Modify the UNSIGNED or SIGNED attribute of a column, for example: INT UNSIGNED -> INT SIGNED

Before TiDB v7.1.0, TiCDC synchronizes DML events with identical old and new data to the downstream. When the downstream is MySQL, these DML events do not cause any data changes until the downstream receives and executes the DDL statement. However, when the downstream is Kafka or cloud storage, TiCDC writes a row of redundant data to the downstream.

Starting from TiDB v7.1.0, TiCDC eliminates these redundant DML events and no longer synchronizes them to downstream.
```

> Starting from TiDB v7.1.0, TiCDC eliminates these redundant DML events and no longer synchronizes them to downstream.

Reviewer: ```suggestion
Starting from TiDB v7.1.0, TiCDC eliminates these redundant DML events and no longer replicates them to downstream.
```
